### DISCUSSION TITLE
stream interface在C synthesis產生ii violation

### DISCUSSION THREAD

**Discussion by BarryJu1999** [bol-edu/HLS-SOC-Discussions#75] [Q&A]
如下圖所示synthesis後會產生ii violation，但之後再去執行co-sim和python的模擬都會過驗證，
想請問這問題該如何解決？

另外在.rpt上latency都是?想請問這樣是正常的嗎?

麻煩大家了，謝謝。

![螢幕擷取畫面 2023-09-28 015949](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/4e0a5b4b-b6bf-4f33-9945-23762ce9259a)
![螢幕擷取畫面 2023-09-28 022003](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/0f5192c4-990c-4279-8c67-0793b38dc7f1)

**Comment by stage0831** [comment]
II violation 中的 II是指iteration interval，會有這個問題是HLS在遇到loop 時會預設要將其優化到II=1，沒有的話就會報出violation。這個問題似乎是不影響電路的功能，若要解決問題可以看一下下面這篇文章

https://zhuanlan.zhihu.com/p/447488933?utm_id=0

**Comment by Fossum2523** [comment]
當loop循環上限是變動的時候，HLS 沒辦法對迴圈做出優化，可以運用 tripcount 指令，或者將上限定義為 C 中的最大值來解決此問題。

你可以參考下圖
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/131112486/559ec2fc-2bd0-415d-8db0-610bb245a152)

**Comment by BarryJu1999** [comment]
了解，謝謝以上兩位的回答幫助我很多，十分感謝。
